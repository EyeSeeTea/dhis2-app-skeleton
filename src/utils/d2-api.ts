import { D2Api } from "../types/d2-api";

interface Instance {
    url: string;
    username?: string;
    password?: string;
}

export function getD2APiFromInstance(instance: Instance) {
    if (instance.username && instance.password) {
        return new D2Api({
            baseUrl: instance.url,
            auth: { username: instance.username, password: instance.password },
            backend: "fetch",
        });
    }

    return new D2Api({ baseUrl: instance.url, backend: "fetch" });
}
