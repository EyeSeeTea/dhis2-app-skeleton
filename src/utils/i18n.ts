import { getModuleForNamespace } from "./i18n-typed";

export default getModuleForNamespace("dhis2-skeleton-app");
