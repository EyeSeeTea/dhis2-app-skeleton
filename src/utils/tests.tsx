import { render, RenderResult } from "@testing-library/react";
import { SnackbarProvider } from "@eyeseetea/d2-ui-components";
import { ReactNode } from "react";
import { AppContext, AppContextState } from "$/webapp/contexts/app-context";
import { getTestCompositionRoot } from "$/CompositionRoot";
import { createAdminUser } from "$/domain/entities/__tests__/userFixtures";

export function getTestContext() {
    const context: AppContextState = {
        currentUser: createAdminUser(),
        compositionRoot: getTestCompositionRoot(),
    };

    return context;
}

export function getReactComponent(children: ReactNode): RenderResult {
    const context = getTestContext();

    return render(
        <AppContext.Provider value={context}>
            <SnackbarProvider>{children}</SnackbarProvider>
        </AppContext.Provider>
    );
}
