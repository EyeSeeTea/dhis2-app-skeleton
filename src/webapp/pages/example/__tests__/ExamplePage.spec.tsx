import { getReactComponent } from "$/utils/tests";
import { ExamplePage } from "$/webapp/pages/example/ExamplePage";

describe("ExamplePage", () => {
    it("renders the feedback component", async () => {
        const view = getView();

        expect(await view.findByText("Hello Mary")).toBeInTheDocument();
        expect(view.asFragment()).toMatchSnapshot();
    });
});

function getView() {
    return getReactComponent(<ExamplePage name="Mary" />);
}
