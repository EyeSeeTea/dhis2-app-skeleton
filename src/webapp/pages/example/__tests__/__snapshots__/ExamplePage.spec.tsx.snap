// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExamplePage > renders the feedback component 1`] = `
<DocumentFragment>
  <div>
    <button
      aria-label="Back"
      class="MuiButtonBase-root MuiIconButton-root sc-gEvDqW sc-eqUzNf hwoczq haqPoX MuiIconButton-colorSecondary"
      data-test="page-header-back"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root MuiIcon-colorPrimary"
        >
          arrow_back
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <h5
      class="MuiTypography-root sc-aYaIB hZQTlr MuiTypography-h5 MuiTypography-gutterBottom"
      data-test="page-header-title"
    >
      Detail page
    </h5>
  </div>
  <h2
    class="sc-fqkwJk lcQFEV"
  >
    Hello Mary
  </h2>
</DocumentFragment>
`;
